src/App.jsx

```diff
@@ -1,36 +1,172 @@
+      const pets = batch.response;
+      this.setState({ pets, page: 1, loading: false });
+    } catch (err) {
+      // Handle error
+      this.openNotification();
+    }
+  };
+
+  handleLoad = async () => {
+    // Prepare to load
+    let pets = this.state.pets;
+    this.setState({
+      loading: true
+    });
+    // Get a new batch
+    try {
+      const newBatch = await getPetBatch(
+        20,
+        this.state.page,
+        this.state.sortBy,
+        this.state.filterBy
+      );
+      // Done displaying
+      if (newBatch.status === false) {
+        message.warning("That's all!");
+        this.setState({
+          hasMore: false,
+          loading: false
+        });
+        return;
+      }
+      pets = [...pets, ...newBatch.response];
+      const newPage = this.state.page + 1;
+      this.setState({ pets, page: newPage, loading: false });
+    } catch (err) {
+      this.openNotification();
+    }
+  };
+
+  closeNotification = key => {
+    notification.close(key);
+    // Go back to loading
+    this.setState({ loading: false });
+  };
+
+  openNotification = () => {
+    const key = `open${Date.now()}`;
+    const btn = (
+      <Button
+        type="primary"
+        size="small"
+        onClick={() => this.closeNotification(key)}
+      >
+        Try Again
+      </Button>
+    );
+    notification.open({
+      message: "Error",
+      description: "Having trouble loading more pets.",
+      btn,
+      key,
+      duration: 0
+    });
+  };
+
+  initiateNameSort = () => {
+    this.setState({
+      pets: [],
+      page: 0,
+      hasMore: true,
+      loading: false,
+      sortBy: "name"
+    });
+  };
+
+  initiateBreedSort = () => {
+    this.setState({
+      pets: [],
+      page: 0,
+      hasMore: true,
+      loading: false,
+      sortBy: "breed"
+    });
+  };
+
+  filterDog = type => {
+    this.setState({
+      pets: [],
+      page: 0,
+      hasMore: true,
+      loading: false,
+      sortBy: null,
+      filterBy: "dog"
+    });
+  };
+
+  filterCat = () => {
+    this.setState({
+      pets: [],
+      page: 0,
+      hasMore: true,
+      loading: false,
+      sortBy: null,
+      filterBy: "cat"
+    });
+  };
+
+  noFilter = () => {
+    this.setState({
+      pets: [],
+      page: 0,
+      hasMore: true,
+      loading: false,
+      sortBy: null,
+      filterBy: null
+    });
+  };
 
   render() {
-    const {
-      pets,
-    } = this.state;
+    const { pets } = this.state;
 
     return (
-      <div className="App">
-        <div className="tableheader">
-          <div className="type">Type</div>
-          <div className="name">Name</div>
-          <div className="breed">Breed</div>
+      <React.Fragment>
+        <h1>Dawgs and Kittehs</h1>
+        <div className="infinite-container">
+          <InfiniteScroll
+            initialLoad={true}
+            pageStart={0}
+            loadMore={page => this.handleLoad(page)}
```

Comment:
page is unused - looks like you're updating it post-server-fetch

Reply:
Yeah, it's a built-in argument from the InfiniteScroll library - I was using it at first, but later I started storing my own in `this.state` because it won't let you reset it if the user sorts/filters.

Reply:
ah, makes sense!